lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -188,11 +188,36 @@ defmodule ExUnit.Runner do
       {:error, %{test_case | state: failed}}
   end
 
-  defp run_test(config, test, context) do
+  defp run_test(true, config, test, context) do
+    run_test(:stdio, config, test, context)
+  end
+
+  defp run_test(false, config, test, context) do
+    spawn_test(config, test, context)
+  end
+
+  defp run_test(device, config, test, context) do
+    ref = make_ref()
+    try do
+      ExUnit.CaptureLog.capture_log(device, fn ->
+        send self(), {ref, spawn_test(config, test, context)}
+      end)
+    catch
+      :exit, :noproc -> %{test | state: {:invalid, :no_logger}}
```

Comment:
Ok, so this means we need to use failed or improve valid. That's because the invalid tests are tagged as so they did not even run because something else ran, failed and reported. For example, today, the second tuple element is a test case: https://github.com/lexmag/elixir/blob/opportunistic-logging-pr/lib/ex_unit/lib/ex_unit/runner.ex#L160

So let's go with failed and raise an exception. We have some other examples elsewhere.